plugin/core/views.py

```diff
@@ -802,7 +807,23 @@ def format_completion(
         st_trigger = lsp_filter_text
         if st_details:
             st_details += " | "
-        st_details += "<p>{}</p>".format(html.escape(lsp_label))
+        if lsp_label_details:
+            lsp_label_detail = lsp_label_details.get("detail")
+            lsp_label_description = lsp_label_details.get("description")
+            st_details += "<p>"
+            # `label` should be rendered most prominent.
+            st_details += _wrap_in_tags("b", lsp_label)
+            if isinstance(lsp_label_detail, str):
+                # `detail` should be rendered less prominent than `label`.
+                # The string is appended directly after `label`, with no additional white space applied.
+                st_details += html.escape(lsp_label_detail)
+            if isinstance(lsp_label_description, str):
+                # `description` should be rendered less prominent than `detail`.
+                # Additional separation is added.
+                st_details += " - {}".format(_wrap_in_tags("i", lsp_label_description))
+            st_details += "</p>"
+        else:
+            st_details += _wrap_in_tags("p", lsp_label)
```

Comment:
One more thing... This logic is only executed in a quite specific case when "lsp_filter_text and lsp_filter_text != lsp_label". I think that this should be supported even when there is no filter text or filter text is not different than label.

Reply:
It looks like we always want to put it into the st_detail slot. The st_annotation slot is reserved for lsp_details.

If there is no lsp_details information, but there are labelDetails, should we put the label details in the st_annotation? I could imagine this to be somewhat confusing if some completions do have an lsp_details and some don't. But, I'm open to any kind of improvement.

Reply:
In short, I think it's fine as it is now (minus the issue below).

> If there is no lsp_details information, but there are labelDetails, should we put the label details in the st_annotation?

With `st_annotation` we can't avoid the spacing between the label and `labelDetails.detail` so I think that is not a good fit.